app/debrief-sections/tactic_technique_table.py

```diff
@@ -12,7 +16,58 @@ def __init__(self):
         self.display_name = 'Tactic and Technique Table'
         self.section_title = 'TACTICS AND TECHNIQUES'
         self.description = ''
+        self.log = logging.getLogger('debrief')
+        self._a18 = None  # lazy-loaded ATT&CK v18 index
+
+    # ---------- ATT&CK v18 Load ----------
+    def _load_attack18(self):
+        """Load ATT&CK v18 index (cached file, works offline)."""
+        if self._a18 is None:
+            cache_path = CACHE_PATH
+            if not os.path.exists(cache_path):
+                cache_path = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'uploads', 'attack18_cache.json'))
+            # Use the shared loader utility
+            bundle = load_attack18_cache(cache_path)
+            self._a18 = Attack18Map(index_bundle(bundle))
+        return self._a18
+
+    def _detection_ids_for_ptid(self, ptid: str, tid: str = None):
+        """
+        Return sorted, unique DET-#### IDs. Prefer exact technique (tid) if provided,
+        then fall back to parent (ptid). Fallback to object-id tail when no DET ext id.
+        """
+        a18 = self._load_attack18()
+        
+        # DEBUG logging
+        for tid in ["T1074.001","T1005","T1560.001","T1041"]:
+            self.log.info("TID %s strategies=%d analytics=%d",
+                        tid, len(a18.get_strategies(tid)), len(a18.get_analytics(tid, platform="windows")))
+        ####
+
+        def _fetch(tid_like: str):
+            ids = []
+            strategies = a18.get_strategies(tid_like) or []
+            for s in strategies:
+                det_id = None
+                for er in (s.get('external_references') or []):
+                    ext = er.get('external_id', '')
+                    if isinstance(ext, str) and ext.startswith('DET-'):
+                        det_id = ext
+                        break
+                if not det_id:
+                    det_id = (s.get('id', '') or '')[-8:] or ''
+                if det_id:
+                    ids.append(det_id)
+            return ids
 
+        out = []
+        if tid:
+            out.extend(_fetch(tid))
+        out.extend(_fetch(ptid))
```

Comment:
@deacon-mp  do we want det IDs for both the subtechnique and parent technique if the subtechnique is provided?